osclib/core.py

```diff
@@ -419,8 +419,10 @@ def attribute_value_delete(apiurl, project, name, namespace='OSRT', package=None
         apiurl, list(filter(None, ['source', project, package, '_attribute', namespace + ':' + name]))))
 
 @memoize(session=True)
-def repository_path_expand(apiurl, project, repo):
+def repository_path_expand(apiurl, project, repo, visited_repos=None):
     """Recursively list underlying projects."""
+    if visited_repos is None:
+        visited_repos = set()
     repos = [[project, repo]]
     meta = ET.fromstringlist(show_project_meta(apiurl, project))
```

Comment:
To also avoid visiting the toplevel project, this should be added here already, and can then be removed in line 439:

```
    if visited_repos is None:
        visited_repos = set()
    visited_repos.add((project, repo))
    repos = [[project, repo]]
    meta = ET.fromstringlist(show_project_meta(apiurl, project))
```

Reply:
Actually, OBS doesn't allow this, which is understandable. So this doesn't really make any difference.